tests/system/test_to_gbq.py

```diff
@@ -118,53 +120,89 @@ def test_series_round_trip(
     (
         pandas.DataFrame(
             {
+                "row_num": [0, 1, 2],
                 "date_col": pandas.Series(
                     ["2021-04-17", "1999-12-31", "2038-01-19"], dtype="datetime64[ns]",
                 ),
             }
         ),
+        None,
         [{"name": "date_col", "type": "DATE"}],
         True,
     ),
+    # Loading a DATE column should work for string objects. See:
+    # https://github.com/googleapis/python-bigquery-pandas/issues/421
+    (
+        pandas.DataFrame(
+            {"row_num": [123], "date_col": ["2021-12-12"]},
+            columns=["row_num", "date_col"],
+        ),
+        pandas.DataFrame(
+            {"row_num": [123], "date_col": [datetime.date(2021, 12, 12)]},
+            columns=["row_num", "date_col"],
+        ),
+        [{"name": "row_num", "type": "INTEGER"}, {"name": "date_col", "type": "DATE"}],
+        False,
+    ),
+    # Loading a NUMERIC column should work for floating point objects. See:
+    # https://github.com/googleapis/python-bigquery-pandas/issues/421
+    (
+        pandas.DataFrame(
+            {"row_num": [123], "num_col": [1.25]}, columns=["row_num", "num_col"],
+        ),
+        pandas.DataFrame(
+            {"row_num": [123], "num_col": [decimal.Decimal("1.25")]},
+            columns=["row_num", "num_col"],
+        ),
+        [
+            {"name": "row_num", "type": "INTEGER"},
+            {"name": "num_col", "type": "NUMERIC"},
+        ],
+        False,
+    ),
 ]
+
 if db_dtypes is not None:
     DATAFRAME_ROUND_TRIPS.append(
         (
             pandas.DataFrame(
                 {
+                    "row_num": [0, 1, 2],
                     "date_col": pandas.Series(
                         ["2021-04-17", "1999-12-31", "2038-01-19"], dtype="dbdate",
                     ),
                 }
             ),
+            None,
```

Comment:
Why is `expected_df` None here? I'm probably missing something.

**Edit:** I see below, it's implicitly saying that it should bee the same as `input_df`.

We might want to consider making these parametrized cases more understandable on the first pass already. Two ideas:
 - Using namedtuples, giving the tuple elements names through the kwarg on nametuple creation.
 - Adding an explanatory comment next to the constants (`None`, `False`...) used in the branching inside the test. Right now one needs to scroll down into the test body to make sense of them.

Reply:
I like the namedtuple idea